app/models/diagnosis.rb

```diff
@@ -19,7 +19,11 @@ class Diagnosis < ApplicationRecord
   scope :of_user, (->(user) { joins(:visit).where(visits: { advisor: user }) })
   scope :reverse_chronological, (-> { order(created_at: :desc) })
   scope :in_progress, (-> { where(step: [1..4]) })
```

Comment:
Maybe change that to `[1..(LAST_STEP _1)]` to be consistent with the completed scope